multiple patches for table population for mantid interface

Kvieta1990 opened the pull request:
A single shot aiming at solving multiple issues with table population/manipulation for mantid interface, as detailed below.

1. Fix the bug with fetching information from oncat.

2. When typing in some text in the search box alongside `List of all runs selected`, the table below would be updated accordingly. In this case, we do expect that runs listed explicitly in the table (after the application of the quick filter specified in the search box) are to be imported when clicking on the `Import` button at the bottom. However, this was not the case and here we intend to implement such an expected behavior.

    2.1. The original `Import All Runs` button text is changed to ` Import All Listed Runs`.

    2.2. When clicking on the `Import All Listed Runs`, only runs that are listed explicitly will be imported.

3. Fix several issues with the right click menu in the main ADDIE processing window (the `Mantid` interface).

    3.1. When right click on the white space (i.e. no cells/rows selected), followed by executing copy/paste/duplicate, etc., ADDIE would crash, complaining about index out of range.

    3.2. When selecting the `Show All Columns` option under `Columns Configuration`, ADDIE would crash. The reason for such a crash is that the dictionary for storing the full configuration is not initialized.

    3.3. When a certain row is copied, more robust behavior is expected. For example, after the copy operation, we can insert a new row and paste the copied row into the new blank row. Though this can be realized by simply duplicating a certain row, still the behavior described here is what a normal user would expect, in terms of the copy/paste operations.

4. Little bug fix - some button icons fail to load at first launching.

marshallmcdonnell commented on addie/processing/mantid/master_table/selection_handler.py:
```diff
@@ -294,6 +306,14 @@ def paste(self, row=None):
             list_to_row = self.o_selection.get_list_row()
+            # if len(list_to_row) == 0:
+            #     print("[Info] No row(s) selected! Highlight any cell(s) in row(s) to remove followed by right click.")
+            #     return
```
We can delete the commented code out, just to clean up a bit.

marshallmcdonnell commented on addie/processing/mantid/master_table/table_tree_handler.py:
```diff
@@ -795,14 +813,24 @@ def rows_remove(self):
-        o_table_row = TableRowHandler(main_window=self.main_window)
-        o_table_row.insert_blank_row()
+        # o_table_row = TableRowHandler(main_window=self.main_window)
+        # o_table_row.insert_blank_row()
```
Same, let's just delete out the old commented out code. We have it in version control if we need it again.

marshallmcdonnell reviewed (changes requested):
Great additions, thank you!
Lot's of great error message additions in the UI and bug fixes.
I haven't tested but will assume all is fine.

I do have two small comments to delete out commented code, just to clean up.
After that, happy to approve!

Kvieta1990 commented:
I rebased my legacy branch onto master and removed those unnecessary comments. It will be very nice if you can help checking and merging. Thank you so much @marshallmcdonnell. But no hurry at all. Please take your time.

marshallmcdonnell reviewed (approved):
LGTM
